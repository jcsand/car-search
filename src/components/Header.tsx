/** @jsxImportSource @emotion/react */
import React from "react";
import { color, ColorProps } from "styled-system";
import styled from "@emotion/styled";
import { css } from "@emotion/react";
import facepaint from "facepaint";
import { Box, Flex } from "reflexbox";

import { Button } from "@@components/Button";
import menuIcon from "@@icons/menu.svg";

// TODO: think about importing these properly
const RENTAL_CARS_LOGO =
  "https://cdn.rcstatic.com/images/site_graphics/newsite/mobile/logos/rc-logo-small--white.svg";

const GBP_FLAG =
  "https://cf.bstatic.com/static/img/flags/new/48-squared/gb.png";

const mq = facepaint([
  // "@media(min-width: 480px)",
  "@media(min-width: 576px)"
  // "@media(min-width: 768px)",
  // "@media(min-width: 1024px)"
  // "@media(only screen and (min-width: 1024px))"
]);

const StyledHeader = styled.header<ColorProps>`
  ${color}
`;

const HeaderFlexContainer = styled.div`
  display: flex;
  align-items: center;
  max-width: 1142px;
  height: 64px;
  margin: 0 auto;
  padding-bottom: calc(4px * 2);
  padding-left: calc(4px * 2);

  ${css(
    mq({
      paddingRight: ["0", "calc(4px * 4)"]
    })
  )}
`;

const HeaderLogo = styled.img`
  display: block;
  margin-left: calc(4px * 2);
  padding-top: calc(4px * 3);
  padding-bottom: calc(4px * 3);
  border-style: none;
`;

const LanguageButton = styled(Button)`
  margin-left: calc(4px * 2);
  background: none;

  ${css(
    mq({
      display: ["none", "block"]
    })
  )}
`;

const LanguageFlagImage = styled.img`
  display: block;
  width: calc(4px * 6);
  height: calc(4px * 6);
  border-radius: 50%;
`;

const HeaderManageBookingButton = styled(Button)`
  min-height: 44px;
  padding: calc(4px * 2) calc(4px * 4);
  border-radius: 4px;
  border-color: currentcolor;
  background-color: #fff;
  color: #1273c4;
  line-height: calc(4px + 44px / 2);
  text-decoration: none;

  &:hover,
  &:focus {
    background-color: #1273c40f;
    background-image: linear-gradient(#1273c40f, #1273c40f),
      linear-gradient(#fff, #fff);
  }

  ${css(
    mq({
      display: ["none", "block"]
    })
  )}
`;

const MenuButton = styled(Button)`
  margin-left: calc(4px * 2);
  background: none;

  ${css(
    mq({
      display: ["block", "none"]
    })
  )}
`;

const MenuImage = styled.img`
  display: block;
  width: calc(4px * 5);
  height: calc(4px * 5);
`;

export const Header: React.FC = () => (
  <StyledHeader color="foreground" bg="background">
    <HeaderFlexContainer>
      <Box flexGrow={1}>
        <a href="/">
          <HeaderLogo alt="Rentalcars.com Brand Logo" src={RENTAL_CARS_LOGO} />
        </a>
      </Box>
      <Box>
        <Flex>
          <LanguageButton>GBP</LanguageButton>
          <LanguageButton>
            <LanguageFlagImage alt="English Language Icon" src={GBP_FLAG} />
          </LanguageButton>
          <MenuButton>
            <MenuImage alt="Menu Icon" src={menuIcon} />
          </MenuButton>
        </Flex>
      </Box>
      <Box ml="calc(4px * 2)" pl="calc(4px * 2)">
        <HeaderManageBookingButton>Manage booking</HeaderManageBookingButton>
      </Box>
    </HeaderFlexContainer>
  </StyledHeader>
);

// <div
//   data-testid="input-text-searchbox-toolbox-fts-end-slot"
//   class="bgui-c-input-text__side bgui-c-input-text__side--end"
// >
//   <span
//     data-testid="searchbox-toolbox-fts-input-loading-spinner"
//     class="bgui-c-spinner bgui-c-spinner--medium"
//   ></span>
// </div>;
