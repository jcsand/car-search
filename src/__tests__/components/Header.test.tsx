import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Header } from "@@components/Header";

describe("Header component", () => {
  it("renders matching snapshot", async () => {
    const rendered = render(<Header />);
    expect(rendered.baseElement).toMatchSnapshot();
  });
});
