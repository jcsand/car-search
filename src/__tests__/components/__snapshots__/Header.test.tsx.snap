// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders matching snapshot 1`] = `
<body>
  <div>
    <header
      class="css-xgd2kp"
      color="foreground"
    >
      <div
        class="css-ik3yhu"
      >
        <div
          class="css-wcquo4"
        >
          <a
            class="css-2b6wk5"
            href="/"
          >
            <img
              alt="Rentalcars.com Brand Logo"
              class="css-1syunqi"
              src="test-file-stub"
            />
          </a>
        </div>
        <div
          class="css-vurnku"
        >
          <div
            class="css-4cffwv"
          >
            <button
              class="css-1kju424"
            >
              GBP
            </button>
            <button
              class="css-1kju424"
            >
              <img
                alt="English Language Icon"
                class="css-1aqz85s"
                src="test-file-stub"
              />
            </button>
            <button
              class="css-13ak2ob"
            >
              <img
                alt="Menu Icon"
                class="css-rsrytf"
                src="test-file-stub"
              />
            </button>
          </div>
        </div>
        <div
          class="css-1j2g464"
        >
          <button
            class="css-1a6r96u"
          >
            Manage booking
          </button>
        </div>
      </div>
    </header>
  </div>
</body>
`;
